### PR TITLE
[generators] add harfbuzz languages generator

### PR DESCRIPTION
**Pull request by benoitkugler**
Add the code generator used for [the harfbuzz language table](https://github.com/go-text/typesetting/blob/main/harfbuzz/ot_language_table.go).

It is compatible with the simplification introduced in https://github.com/go-text/typesetting/pull/66

### CODE REVIEW

**Review by whereswaldon — APPROVED** [review]
Again, I have to applaud a colossal effort. Thank you for tackling this. :D I've read over the generator code and it seems reasonable.

File: generators/langs/gen.go [hunk 1]
```diff
@@ -0,0 +1,1500 @@
...
+// which has the generic form <lang>-<country>-<other tags>...
+type Language string
+
+// NewLanguage canonicalizes the language input (as a BCP 47 language tag), by converting it to
+// lowercase, mapping '_' to '-', and stripping all characters other
+// than letters, numbers and '-'.
+func NewLanguage(language string) Language {
+	out := make([]byte, 0, len(language))
+	for _, b := range language {
+		can := canonMap[b]
```
Comment (andydotxyz) [inline generators/langs/gen.go]: Won't this panic if the language (which is user provided) have unicode or upper ascii characters?

**Review by andydotxyz — COMMENTED** [review]
A little question about possible panic.
Not sure if that is an issue when it's in a generator but thought I would ask in case.

File: generators/langs/gen.go (hunk 1 above)
Comment (benoitkugler) [reply generators/langs/gen.go]: Indeed, good catch ! I've had the typical byte/rune confusion...
I've opened https://github.com/go-text/typesetting/pull/67 to "backport" the fix.

**Review by andydotxyz — APPROVED** [review]
Nice thanks

**Pull request merged**